tests/test_utils.py

```diff
@@ -151,6 +153,9 @@ def validate_uuid4(uuid_string: str) -> bool:
 
 
 def get_sqlite_temp_table_names(engine):
+    if isinstance(engine, Engine):
+        engine = engine.connect()
+
     result = engine.execute(
```

Comment:
Let's call this connection

```suggestion
        connection = engine.connect()

    result = connection.execute(
```